### DISCUSSION TITLE
[CHORE] How to set variable parameters for IF activities in Elsa 3.1.3 designer?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5717] [Q&A]
How to set variable parameters for IF activities in Elsa 3.1.3 designer?

**Comment by glime-ai** [comment]
To set variable parameters for IF activities in Elsa 3.1.3 designer, you would typically use the Condition attribute to determine the flow of the workflow. Based on the information provided in the context, the IF activity in Elsa 3.1.3 designer shows two buttons in the activity display for "Then" and "Else" which represent sub-flowcharts. You can click on these buttons to go into a sub-flowchart where you can implement your logic by dragging and dropping new activities.

Here's a step-by-step guide on how to set variable parameters for IF activities:

1. Add an IF activity to your workflow in the Elsa designer.
2. Click on the IF activity to configure it.
3. Set the Condition attribute with an expression that evaluates to a boolean value.
4. Click on the "Then" button to define the activities that should be executed when the condition is true. This will open a sub-flowchart where you can add and connect activities.
5. Similarly, click on the "Else" button to define the activities for when the condition is false.
6. Once you have configured the "Then" and "Else" branches, you can connect the "Done" connection to subsequent activities in your workflow.

Please note that if you encounter any issues with the visibility of the Then and Else properties in the designer interface, as mentioned in the context, you may need to check for updates or report the issue to the Elsa GitHub repository as it might be a bug.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by sfmskywalker** [comment]
You can configure the If activity using the designer by specifying an expression for its _Condition_ field:

<img width="1968" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/938393/e2c8b027-76c4-4b3d-9604-83bb80301867">

This expression can include references to workflow variables, as seen from the screenshot.

**Reply by shxylgithub** [reply]
Thank you, teacher

**Comment by glime-ai** [comment]
To set variable parameters for IF activities in Elsa 3.1.3 designer, you would typically use the designer interface to configure the properties of the IF activity, including the condition that determines whether the 'Then' or 'Else' branch should be executed. However, based on the context provided, there seems to be an issue or a lack of clarity on how to do this directly in the Elsa 3.1.3 designer.

As of my knowledge cutoff date in early 2023, the Elsa documentation or the GitHub issues do not provide a direct answer to this question. Therefore, I would recommend checking the official Elsa documentation for any updates or looking for community support on platforms like Stack Overflow or the Elsa GitHub repository issues section.

If the designer interface does not expose the necessary properties to set variable parameters, you may need to define the condition programmatically or look for a workaround. It's also possible that this could be a bug or a missing feature in the version you are using, in which case you might want to report it to the Elsa GitHub issues page for further assistance.

Since the context does not provide a clear solution, I would suggest reaching out to the Elsa community or maintainers for guidance on this specific version.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
